src/AbstractPathMappingRepository.php

```diff
@@ -79,21 +73,61 @@ public function add($path, $resource)
             return;
         }
 
-        if ($resource instanceof FilesystemResource) {
-            $this->ensureDirectoryExists(Path::getDirectory($path));
-            $this->addResource($path, $resource);
-            $this->sortStore();
+        $this->ensureDirectoryExists(Path::getDirectory($path));
+        $this->addResource($path, $resource);
+        $this->sortStore();
+    }
 
-            return;
+    /**
+     * Add the resource (internal method after checks of add()).
+     *
+     * @param string   $path
+     * @param Resource $resource
+     */
+    private function addResource($path, $resource)
+    {
+        if ($resource instanceof Resource) {
```

Comment:
You can invert the condition and throw the exception inside to reduce the code nesting level.